stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/actions/Close.java

```diff
@@ -42,6 +44,17 @@
 
     @Override
     public void execute(Context context) {
-        closeConnections(retrieveDatasource(context));
+        SQLDatasource datasource = retrieveDatasource(context);
+        // When an exception is thrown during database endpoint init (eg: driver not present) stop operation
+        // of the endpoint is automatically called. But at this point, datasource is null therefore to handle that
+        // situation following null check is needed.
+        if (datasource != null && !datasource.isGlobalDatasource()) {
+            try {
+                datasource.decrementClientCounter();
+            } catch (InterruptedException e) {
+                context.setReturnValues(
+                        SQLDatasourceUtils.getSQLConnectorError(context, "Error while stopping the database client"));
```

Comment:
Shall we add more information to the error message based on the message in InterruptedException?

Reply:
If that happens, I doubt how much a ballerina developer can do about a java thread interruption.
Also, in this case, the InterruptedException object doesn't have a message so all we know is thread was interrupted while waiting for the mutex, hence the exception.